src/widgets/fileview.cpp

```diff
@@ -268,6 +270,7 @@ void FileView::showEvent(QShowEvent *e) {
   model_ = new QFileSystemModel(this);
 
   model_->setNameFilters(filter_list_);
+  model_->setIconProvider(new QFileIconProvider());
```

Comment:
This creates a memory leak. Why is it necessary to construct a new `QFileIconProvider` in the first place? I would think it should be possible to set options on the existing one accessed with `model->iconProvider()`

Reply:
> This creates a memory leak.

Hm, how is that? :thinking: 
This should only be created once, due to the  `if (model_) return;` a few lines before.
(#noirony: I usually don't do stuff in C++ very often, let alone QT - so maybe I missed something)


> Why is it necessary to construct a new QFileIconProvider in the first place?

A reasonable question. The only Option-Flag I found in the [documentation](https://doc.qt.io/qt-6/qabstractfileiconprovider.html#Option-enum) was `DontUseCustomDirectoryIcons` - and it had no effect.

But thinking about it: it might be a good idea, to check if the current IconProvider is usable, before setting a new one.
Something like `provider->icon(QAbstractFileIconProvider::Folder).isNull()`

...will prepare an update for the PR.
But any other suggestions wouldn't go amiss.

Reply:
> > This creates a memory leak.
> 
> Hm, how is that? thinking This should only be created once, due to the `if (model_) return;` a few lines before. (#noirony: I usually don't do stuff in C++ very often, let alone QT - so maybe I missed something)
> 
Because the memory should be fee'd on exit.
Use `std::unique_ptr`